From 56a0aac777ef2db95d27d214bf7577be64a44c23 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Sat, 28 May 2022 20:39:01 +0200
Subject: [PATCH 3/6] clockmodel.cpp: Move away from deprecated toTime_t

toTime_t() has been deprecated as per https://doc.qt.io/qt-5/qdatetime-obsolete.html#fromTime_t-1

In order to keep backwards compatibility with Qt 5.6 used by SFOS, use toMSecsSinceEpoch instead.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
 libconnman-qt/clockmodel.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libconnman-qt/clockmodel.cpp b/libconnman-qt/clockmodel.cpp
index 72dbb53..4154064 100644
--- a/libconnman-qt/clockmodel.cpp
+++ b/libconnman-qt/clockmodel.cpp
@@ -154,13 +154,13 @@ void ClockModel::setTimeservers(const QStringList &val)
 void ClockModel::setDate(QDate date)
 {
     QDateTime toDate(date, QTime::currentTime());
-    quint64 secsSinceEpoch = (quint64)toDate.toTime_t();
+    quint64 secsSinceEpoch = (quint64)toDate.toMSecsSinceEpoch() / 1000;
     SET_CONNMAN_PROPERTY("Time", secsSinceEpoch);
 }
 
 void ClockModel::setTime(QTime time)
 {
     QDateTime toDate(QDate::currentDate(), time);
-    quint64 secsSinceEpoch = (quint64)toDate.toTime_t();
+    quint64 secsSinceEpoch = (quint64)toDate.toMSecsSinceEpoch() / 1000;
     SET_CONNMAN_PROPERTY("Time", secsSinceEpoch);
 }
-- 
2.36.1

